### PR TITLE
Add basic support for '--targets/-t'

### LINKED ISSUE #16: Basic regions support
We should add the equivalent of the bcftools `-r/--regions` option to filter by regions.

The work in https://github.com/sgkit-dev/sgkit/pull/658 (which was never merged) could form the basis for the implementation.

Also, it would be simpler to implement `-t/--targets` first, since unlike `-r/--regions` it only needs to check position, and not the variant length. To support the latter we probably need to store the variant length as a separate Zarr array in order to do efficient queries.

### PR DESCRIPTION
**Pull request by tomwhite**
Fixes #16

This is just a quick draft, but I wanted to share early for visibility.

* It uses a variant mask, since we will need one anyway for overlap queries. (Some of the code is inspired by https://github.com/sgkit-dev/vcf-zarr-publication/blob/52428c97822a35a7c29f150f35ec6bd7a29dad37/src/zarr_afdist.py)
* I haven't thought about efficiency. The most glaring is that chunks that are masked out are not skipped - perhaps that should be done here. The linked issues in #16 could be looked at later.
* It only handles a very limited part of the `-t` syntax - still needs to handle missing start/end, multiple targets, and the ^ syntax (for exclude).
* I only implemented it for the C implementation branch. It would be easy enough to add for Numba, but perhaps we could drop that branch soon if the C branch has feature parity now?
* It needs tests and perhaps a bit more doc.
* We should add linting to the repo.

### CODE REVIEW

**Comment by jeromekelleher** [comment]
This is great, thanks! I think skipping fully masked out chunks is good enough in the first instance perf wise - otherwise I don't think this will be practical for looking bits of large datasets.

**Comment by tomwhite** [comment]
This addresses most of the things discussed above (skipping fully masked out chunks, adding numba, region start/end, multiple regions).

I also introduced pyranges and found that it made things simpler even for the `--targets` case, and will make the `--regions` option straightforward from here. I had a look at bioframe too, and it is very similar, but I noticed it has a dependency on matplotlib I'd rather avoid if possible.

Still need to add support for the ^ exclude syntax, and more tests for edge cases, but I think these can be added later. (I'm away for the first half of next week BTW.)

File: vcztools/regions.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import re
+from typing import Any, List, Optional
+
+import numpy as np
+import pandas as pd
+import pyranges
+
+
+def parse_region(region: str) -> tuple[str, Optional[int], Optional[int]]:
```
Comment (tomwhite) [inline vcztools/regions.py:9]: Ideally this would return a `Region` object, like the one in bio2zarr, but I'm not sure which way the dependency is between the two projects (if indeed there is one). So I left it as a tuple for now.

**Comment by jeromekelleher** [comment]
Thanks @tomwhite, this is great. 

I'd be pretty slow to add external dependencies here, and I'm sceptical about whether we're really getting much value out of pyranges and pandas here for what we're currently doing. But, I'd vote we merge this and use it as a data point, and see if doing things with a simpler approach can achieve as good or better performance.

**Comment by jeromekelleher** [comment]
So, I'm happy to merge this now, if you are?

**Comment by tomwhite** [comment]
> So, I'm happy to merge this now, if you are?

Yes, please go ahead! (Sorry for slow response, I was away for a few days.)

**Comment by jeromekelleher** [comment]
No worries, I knew you were away
